R/idr0021 - Segmentation.ipynb

```diff
@@ -0,0 +1,702 @@
+    "        y <- features[index,9]\n",
+    "        r1 <- features[index,10]\n",
+    "        ecc <- features[index,11]\n",
+    "        r2 <- sqrt(r1^2 * (1 - ecc^2))\n",
+    "        theta <- features[index,12]\n",
+    "        rois <- rbind(rois, c(x, y, r1, r2, NA, NA, -theta, as.character(index)))\n",
+    "    }\n",
+    "    rois <- rois[-1,]\n",
+    "    rownames(rois) <- c()\n",
+    "    return(rois)\n",
+    "}"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "rois <- createROIs(features)\n",
+    "rois"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Save the ROIs back to OMERO"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "addROIs(image, coords = rois)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**Exercise:** Open the image in OMERO.web and compare the ROIs with the segmented image."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Store the features table"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Attach the whole dataframe to the image:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "invisible(attachDataframe(image, features, \"ROI features\")) # the invisible just suppresses some unneccessary output"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The dataframe is now attached to the image as an [HDF](https://en.wikipedia.org/wiki/Hierarchical_Data_Format) table. You can download and open it with software like HDF Compass, load it into python scripts using 'h5py' library, etc. or simply load it directly as R dataframe again."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Alternatively attach it as CSV file:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "csvFile <- \"/tmp/ROI_features.csv\"\n",
+    "write.csv(features, file = csvFile)\n",
+    "fileannotation <- attachFile(image, csvFile)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**Exercise:** Find the attached CSV file and open it in Excel"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Automate"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Task: Segment each image of a dataset, create an ROI for each of the objects, and summarize the results as dataframe, only taking the features of the largest ROI of an image into account.\n",
+    "\n",
+    "Note: We need to be able to specify the channel which is analyzed by its name.\n",
```

Comment:
``We need to be able to specify the channel which is analyzed by its name.`` -> ``Only one channel in each image is relevant for our analysis. The relevant channel is identified by its name.``

this would be re-written what you have there - but didn't you also want to say that the name of the channel should match the name of the dataset containing the image ?